rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp

```diff
@@ -50,6 +56,7 @@ struct CustomSubscriberInfo : public CustomEventInfo
   const void * type_support_impl_{nullptr};
   rmw_gid_t subscription_gid_{};
   const char * typesupport_identifier_{nullptr};
+  std::shared_ptr<rmw_fastrtps_shared_cpp::LoanManager> loan_manager_;
```

Comment:
@MiguelCompany I know these bits are in a grey area when it comes to code visibility but this isn't ABI compatible, strictly speaking.

Reply:
It is, strictly speaking. But `rmw_fastrtps_shared_cpp` is designed to be only used by `rmw_fastrtps_cpp` and `rmw_fastrtps_dynamic_cpp`, which this PR is also modifying. This means ABI is kept to the users of the RMW.

Reply:
We need a clearer definition of "public" here. Please correct me if I'm mistaken but as I understand it, this will _technically_ break ABI compatibility even if you don't intend for this struct to be part of the formal `rmw_fastrtps*` public API. By that, I mean that if we publish this, and I `apt update ros-rolling-rmw-fastrtps-shared-cpp` without updating the consumers of that package, the consumers won't allocate adequate space for this struct but code in `rmw_fastrtps_shared_cpp` will try to access it anyway.

One can blur the lines of what constitutes "public API" by claiming that not everything exposed in the headers are formally part of the API, but the linker doesn't care what we say. ABI is ABI, and when you change the size of a struct, it is no longer compatible.

I'm going to bring this up at the next ROS 2 meeting so that we can get some solid documentation on the matter and clearly define our goals regarding ABI stability within a rosdistro so that we can come up with some criteria for when we can break it.

Reply:
This new field is added at the end, so offsets of the old fields are not modified. The new field is only being accessed from new functions of `rmw_fastrtps_shared_cpp`. This means that consumers which are not updated will not be calling code accessing the new field.

Reply:
We discussed this at the ROS 2 meeting, and it sounds like stability among arbitrary combinations of package updates isn't part of our ABI stability goal, so my concerns are no longer valid. Thanks for your patience.